### PR TITLE
Add Records with no date in a separate format in condition history

### PR DESCRIPTION
**Pull request by pmadria3**
This PR adds a separate section for records with no date. If there is no entry, then no section will show up. 

With Both Date and No Date: 
Screen Shot 2022-10-06 at 2 16 13 PM

With only Dates:
Screen Shot 2022-10-06 at 2 19 17 PM

With only no Dates: 
Screen Shot 2022-10-06 at 2 20 09 PM

### CODE REVIEW

File: .changeset/quiet-eyes-buy.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@zus-health/ctw-component-library": patch
+---
+
+Condition History has a header that subdivides the condition entries to sort by date otherwise group the entries with no dates.
```
Comment (WesleyKapow) [inline .changeset/quiet-eyes-buy.md:5]: This is a little wordy and confusing. Pretend you are talking to the customer and keep it simple if possible.

File: src/components/content/conditions-history.tsx [hunk 2]
```diff
@@ -102,21 +103,39 @@ export function ConditionHistory({ condition }: { condition: ConditionModel }) {
...
       setLoading(true);
     };
   }, [condition, patientResponse.data, historyResponse.data]);
 
   function conditionHistoryDisplay() {
-    if (conditions.length === 0 && !loading) {
+    if (
+      conditionsWithDate.length === 0 &&
+      !loading &&
+      conditionsWithoutDate.length === 0
```
Comment (WesleyKapow) [inline src/components/content/conditions-history.tsx:137]: nit: Put the two `=== 0 ` checks next to each other.

File: src/components/content/conditions-history.tsx [hunk 3]
```diff
@@ -138,9 +158,18 @@ export function ConditionHistory({ condition }: { condition: ConditionModel }) {
           <div className="ctw-text-sm">{condition.ccsGrouping}</div>
         </div>
         <CollapsibleDataListStack
-          entries={conditions}
+          entries={conditionsWithDate}
           limit={CONDITION_HISTORY_LIMIT}
         />
+        <div className="ctw-space-y-2 ctw-text-base">
+          {conditionsWithoutDate.length !== 0 && (
```
Comment (luthfib) [inline src/components/content/conditions-history.tsx:165]: Please increase the font weight to 500

Comment (luthfib) [reply src/components/content/conditions-history.tsx:165]: Also please match indentation to the figma file for "Records with no date:"

File: src/components/content/conditions-history.tsx [hunk 4]
```diff
@@ -138,9 +158,18 @@ export function ConditionHistory({ condition }: { condition: ConditionModel }) {
...
+            <div className="ctw-space-y-2">Records with no date:</div>
```
Comment (WesleyKapow) [inline src/components/content/conditions-history.tsx:166]: Spacing classes are used to add space between children elements. This only has one child and so will do nothing.

File: .changeset/quiet-eyes-buy.md (hunk 1 above)
Comment (melocule) [inline .changeset/quiet-eyes-buy.md:5]: Should this be a patch or a minor? I read that patches are for bug fixes and minors are for backward-compatible new features

Comment (WesleyKapow) [reply .changeset/quiet-eyes-buy.md:5]: Ya I haven't really pinned down my thinking on this yet. What you read is my understanding as well. We can play this a little loose a.t.m. For most libraries they are able to release patch updates for older versions. We're not setup do that at the moment. We really have linear updates right now (less than ideal but fine for now).

Comment (pmadria3) [reply .changeset/quiet-eyes-buy.md]: I'll leave it as a patch for now? And then I guess moving forward if it's anything more than a bug fix maybe make it a minor? Just for some consistency.

File: src/components/content/conditions-history.tsx [hunk 5]
```diff
@@ -161,14 +161,18 @@ export function ConditionHistory({ condition }: { condition: ConditionModel }) {
           entries={conditionsWithDate}
           limit={CONDITION_HISTORY_LIMIT}
         />
-        <div className="ctw-space-y-2 ctw-text-base">
+        <div className="ctw-space-y-2">
```
Comment (WesleyKapow) [inline src/components/content/conditions-history.tsx:164]: Do we want this div when we are not showing the no date section below?

File: src/components/content/conditions-history.tsx [hunk 6]
```diff
@@ -138,9 +158,22 @@ export function ConditionHistory({ condition }: { condition: ConditionModel }) {
           <div className="ctw-text-sm">{condition.ccsGrouping}</div>
         </div>
         <CollapsibleDataListStack
-          entries={conditions}
+          entries={conditionsWithDate}
           limit={CONDITION_HISTORY_LIMIT}
         />
+        {conditionsWithoutDate.length !== 0 && (
+          <>
```
Comment (WesleyKapow) [inline src/components/content/conditions-history.tsx:165]: And now this fragment is not needed.

Comment (pmadria3) [reply src/components/content/conditions-history.tsx]: ah good catch

**Review by WesleyKapow — APPROVED** [review]

**Pull request merged**